docs/decisions/0002-enterprise-catalog-ida.rst

```diff
@@ -0,0 +1,64 @@
+New Enterprise Catalog IDA
+--------------------------
+
+Status
+======
+
+Draft
+
+Context
+=======
+We are interested in improving the reliability and performance of our system at large, and we have identified that the enterprise catalog api and related functionality are a central dependency that often causes issues, such as users experiencing random failures during enrollment or other workflows. Almost every feature that our users interact with depends on enterprise catalogs: Enrollment, Code Redemption, Data Sharing Consent, Data transmission to external LMS systems, and API access given to select customers.
+
+Our main goals are as follows:
+
+1. Get the response time of the enterprise customer catalog endpoints to be below 1 second for 99% of calls.
+
+2. Since many features do not hit the enterprise catalog api endpoints but perform the same calculations directly in code, we would also like to improve the performance of those code paths and drastically reduce the calls to the discovery service.
```

Comment:
Would it be helpful if an example code path was provided here?

Reply:
sure, i can list some.